modules/ip/control/nexthop.c

```diff
@@ -74,6 +74,17 @@ int ip4_nexthop_add(uint16_t vrf_id, ip4_addr_t ip, uint32_t *idx, struct nextho
 void ip4_nexthop_decref(struct nexthop *nh) {
 	if (nh->ref_count <= 1) {
 		struct nexthop_key key = {nh->ip, nh->vrf_id};
+
+		rte_spinlock_lock(&nh->lock);
+		// Flush all held packets.
+		struct rte_mbuf *m = nh->held_pkts_head;
+		while (m != NULL) {
+			struct rte_mbuf *next = queue_mbuf_data(m)->next;
+			rte_pktmbuf_free(m);
+			m = next;
+		}
+		rte_spinlock_unlock(&nh->lock);
```

Comment:
Can you split this into a separate patch? This seems more like a fix related to nexthops.

Reply:
indeed.